### PR TITLE
Fallback to use the redis command migration type if the target don't support the ApplyBatch command

### LINKED ISSUE #2078: Use redis-command data migration type when the destination does not support ApplyBatch command
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

In https://github.com/apache/kvrocks/pull/2010, we support the new `APPLYBATCH` command for data migration scenarios. In https://github.com/apache/kvrocks/pull/2067, we use this command, to realize the new `raw-key-value` migration type. 

But the `raw-key-value` migration won't work if the target doesn't support the `APPLYBATCH` command, so we can add a fallback mechanism. We can use the old `redis-command` migration type when the destination does not support `ApplyBatch` command.

### Solution

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by caipengbo**
This close #2078

### CODE REVIEW

File: src/cluster/slot_migrate.cc [hunk 1]
```diff
@@ -485,6 +497,33 @@ Status SlotMigrator::setImportStatusOnDstNode(int sock_fd, int status) {
   return Status::OK();
 }
 
+StatusOr<bool> SlotMigrator::supporttedApplyBatchCommandOnDstNode(int sock_fd) {
+  std::string cmd = redis::ArrayOfBulkStrings({"command", "info", "applybatch——"});
+  auto s = util::SockSend(sock_fd, cmd);
+  if (!s.IsOK()) {
+    return s.Prefixed("failed to send command info to the destination node");
+  }
+
```
Comment (caipengbo) [inline src/cluster/slot_migrate.cc:506]: It's not too strict here, maybe we can support redis client lib in the future.

File: src/cluster/slot_migrate.cc [hunk 2]
```diff
@@ -485,6 +497,33 @@ Status SlotMigrator::setImportStatusOnDstNode(int sock_fd, int status) {
   return Status::OK();
 }
 
+StatusOr<bool> SlotMigrator::supportedApplyBatchCommandOnDstNode(int sock_fd) {
+  std::string cmd = redis::ArrayOfBulkStrings({"command", "info", "applybatch——"});
```
Comment (PragmaTwice) [inline src/cluster/slot_migrate.cc]: `applybatch——`?

Comment (caipengbo) [reply src/cluster/slot_migrate.cc]: sorry, for local test :(

**Comment by caipengbo** [comment]
When #2123 is merged (fix a bug about `COMMAND` command), the unit tests will be passed :)

**Review by PragmaTwice — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Pull request merged**